site/en/docs/privacy-sandbox/shared-storage/index.md

```diff
@@ -0,0 +1,219 @@
+
+## Use cases for Shared Storage
+
+The Shared Storage API intends to support many use cases, replacing several
+existing uses for third-party cookies. This may include:
+
+*  Recording aggregated statistics, such as demographics, reach, frequency
+   measurement, and conversion measurement with the
+   [Private Aggregation API](https://github.com/alexmturner/private-aggregation-api)
+*  Frequency capping
+*  Lift experiments
+*  A/B experimentation
+
+The proposal intends to create a general purpose API which supports many
+possible future use cases. This allows for further experimentation and change,
+to grow alongside the web ecosystem.
+
+
+## How will shared storage work?
+
+Shared storage will allow you to make informed decisions based on cross-site
+data, without sharing user information (such as browser history or other
+personal details) with an embedding site. You can write to  shared storage at
+any time, like other JavaScript storage APIs (like localStorage or indexedDB).
+Unlike the other storage APIs, you can only read the shared storage values in
+a secure environment, known as a shared storage worklet.
+
+The shared storage data can be used for:
+
+*  [**URL selection**](#url-selection): you can run a worklet script to select
+   a URL from a provided list, based on the stored data, and then render that
+   URL in a fenced frame.  The returned URL will be an opaque URL, which means
+   the developer and other viewers of the code won't know which URL was
+   selected.
+*  [**Noisy aggregation of cross-site data**](#aggregated-data): you will be
+   able to run a worklet script to send your data through the
+   [Private Aggregation API](https://github.com/alexmturner/private-aggregation-api),
+   a Privacy Sandbox proposal, which returns a privacy-preserving report. 
+
+
+### Example: URL selection for frequency capping {: #url-selection }
+
+To select and create an opaque URL, register a worklet module to read shared
+storage data. The worklet class receives a list of up to eight URLs and then
+returns the index of the chosen URL. 
+
+When the client calls `sharedStorage.runURLSelectionOperation()`, the worklet
+executes and returns an opaque URL to be rendered into a fenced frame.
+
+Let's say you want to render an ad based on the advertiser's frequency cap
+(the maximum number of impressions for a user on a single ad). The frequency
+cap value is stored in shared storage. The shared storage worklet reads the
+values in shared storage, and decrements the value with each additional view.
+If there are available impressions left (the user has not hit their frequency
+cap), the ad is returned (index `1`). If not, the default URL is returned
+(index `0`).
+
+In this example, the adtech creates two files:
+
+*  `get-url.js` is the shared storage worklet, which evaluates the frequency
+   cap to determine which URL is returned for the ad creative.
+*  `main.js` selects and renders the ad creative and is injected on a
```

Comment:
I like how you have `ad-iframe.js` in the second example.  Let's update this to `ad-iframe.js` also to match the second example.